examples/dataflow-csvio/src/main/java/com/google/example/csvio/SortContextualHeadersAndRows.java

```diff
@@ -0,0 +1,427 @@
+          TAG_BASE + "/" + SortMinimumHeaderAndRows.class.getSimpleName(),
+          new SortMinimumHeaderAndRows());
+    }
+    if (configuration.getHeaderMatchRegex() != null) {
+      return nonEmpty.apply(
+          TAG_BASE + "/" + SortMatchedHeaderAndRows.class.getSimpleName(),
+          new SortMatchedHeaderAndRows(configuration));
+    }
+    return nonEmpty.apply(
+        TAG_BASE + "/" + SortFixedHeaderAndRows.class.getSimpleName(),
+        new SortFixedHeaderAndRows(configuration));
+  }
+
+  @AutoValue.Builder
+  abstract static class Builder {
+
+    public abstract Builder setConfiguration(CSVIOReadConfiguration value);
+
+    public abstract SortContextualHeadersAndRows build();
+  }
+
+  /**
+   * The result of processing the raw {@link ContextualTextIO#read()} {@link Row} {@link
+   * PCollection} output into the corresponding CSV headers and rows.
+   */
+  static class Result implements POutput, PInput {
+
+    private final Pipeline pipeline;
+    private final PCollection<Row> headers;
+    private final PCollection<Row> rows;
+
+    Result(PCollectionTuple pct) {
+      this.pipeline = pct.getPipeline();
+      this.headers = pct.get(HEADERS).setRowSchema(RecordWithMetadata.getSchema());
+      this.rows = pct.get(ROWS).setRowSchema(RecordWithMetadata.getSchema());
+    }
+
+    public PCollection<Row> getHeaders() {
+      return headers;
+    }
+
+    public PCollection<Row> getRows() {
+      return rows;
+    }
+
+    @Override
+    public Pipeline getPipeline() {
+      return this.pipeline;
+    }
+
+    @Override
+    public Map<TupleTag<?>, PValue> expand() {
+      return new HashMap<>() {
+        {
+          this.put(HEADERS, headers);
+          this.put(ROWS, rows);
+        }
+      };
+    }
+
+    public CSVIO.Read.Result apply(PTransform<? super Result, CSVIO.Read.Result> t) {
+      return t.expand(this);
+    }
+
+    @Override
+    public void finishSpecifyingOutput(
+        String transformName, PInput input, PTransform<?, ?> transform) {}
+  }
+
+  /**
+   * A {@link PTransform} responsible for bundling the raw {@link ContextualTextIO#read()} {@link
+   * Row} {@link PCollection} output into the corresponding CSV headers and rows in the setting
+   * where a fixed header position is expected.
+   */
+  static class SortFixedHeaderAndRows extends PTransform<PCollection<Row>, Result> {
+
+    private static final String TAG_BASE =
+        SortContextualHeadersAndRows.TAG_BASE + "/" + SortFixedHeaderAndRows.class.getSimpleName();
+
+    private final CSVIOReadConfiguration configuration;
+
+    SortFixedHeaderAndRows(CSVIOReadConfiguration configuration) {
+      this.configuration = configuration;
+    }
+
+    @Override
+    public Result expand(PCollection<Row> input) {
+      PCollectionView<Long> headerPositionView =
+          input
+              .getPipeline()
+              .apply(TAG_BASE + "/HeaderPositionView", Create.of(configuration.getHeaderPosition()))
```

Comment:
apply method can be directly used on Pcollection instead of  Pcollection.getPipeline().apply